lib/git_worktree.rb

```diff
@@ -0,0 +1,336 @@
+require 'rugged'
+
+class GitWorktree
+  attr_accessor :name, :email, :base_name
+  ENTRY_KEYS = [:path, :dev, :ino, :mode, :gid, :uid, :ctime, :mtime]
+  DEFAULT_FILE_MODE = 0100644
+  LOCK_REFERENCE = 'refs/locks'
+  HEAD_REF = 'refs/heads/master'
+
+  def initialize(options = {})
+    raise ArgumentError, "Must specify path" unless options.key?(:path)
+    @path          = options[:path]
+    @email         = options[:email]
+    @name          = options[:name]
+    @bare          = options[:bare]
+    @commit_sha    = options[:commit_sha]
+    @password      = options[:password]
+    @fast_forward_merge = options[:ff] || true
+    @ssl_no_verify = options[:ssl_no_verify] || false
+    @remote_name   = 'origin'
+    @cred          = Rugged::Credentials::UserPassword.new(:username => @name,
```

Comment:
name feels weird as the username.  I think the incoming parameter should be username.